test/base.test.js

```diff
@@ -455,3 +455,41 @@ test('should handle custom signer', async (t) => {
   t.equal(response.headers['set-cookie'].includes(DEFAULT_ENCRYPTED_SESSION_ID), false)
   t.match(response.headers['set-cookie'], /sessionId=[\w-]{32}.[\w-%]{43,135}; Path=\/; HttpOnly; Secure/)
 })
+
+test('should set session cookie if saveUninitialized is false and maxAge is on', async (t) => {
+  t.plan(2)
+  const DateNow = Date.now
+  const now = Date.now()
+  Date.now = () => now
+
+  const options = {
+    cookie: {
+      maxAge: 42
+    },
+    secret: DEFAULT_SECRET,
+    saveUninitialized: false
+  }
+  const plugin = fastifyPlugin(async (fastify, opts) => {
+    fastify.addHook('onRequest', (request, reply, done) => {
+      request.sessionStore.set(DEFAULT_SESSION_ID, {
+        // In this scenario, maxAge would have set expired in a previous request
+        cookie: {
+          expires: new Date(Date.now() + 10)
+        }
+      }, done)
+    })
+  })
+  const fastify = await buildFastify((request, reply) => reply.send(200), options, plugin)
+  t.teardown(() => {
+    fastify.close()
+    Date.now = DateNow
+  })
+
+  const response = await fastify.inject({
+    url: '/',
+    headers: { cookie: DEFAULT_COOKIE, 'x-forwarded-proto': 'https' }
+  })
+
+  t.equal(response.statusCode, 200)
+  t.match(response.headers['set-cookie'], RegExp(`sessionId=.*; Path=/; Expires=${new Date(now + 42).toUTCString()}; HttpOnly`))
```

Comment:
Why is it `+42` instead of `+10`? Should the `sessionStore.set` take priority?

Reply:
Nope, this hook runs before `@fastify/session`. This is the pattern we use in most of our tests to simulate a pre-existing session.